tools/update-undici.sh

```diff
@@ -26,6 +26,13 @@ rm -f deps/undici/undici.js
     "$NODE" "$NPM" install --global-style --no-bin-links --ignore-scripts undici
     cd node_modules/undici
     "$NODE" "$NPM" run build:node
+    # get the new version of undici
+    UNDICI_VERSION=$("$NODE" -p "require('./package.json').version")
+    # update this version information in src/node_metadata.cc
+    sed -i '' "s/UNDICI_VERSION \"[0-9.]*\"/UNDICI_VERSION \"$UNDICI_VERSION\"/" "$ROOT/src/node_metadata.cc"
```

Comment:
`sed -i` is a GNU-only feature, it won't work on non-GNU`sed`. Can we use something more portable? E.g. using git:
```suggestion
    git show "HEAD:$ROOT/src/node_metadata.cc" | \
      sed "s/UNDICI_VERSION \"[0-9.]*\"/UNDICI_VERSION \"$UNDICI_VERSION\"/" \
      > "$ROOT/src/node_metadata.cc"
```

Reply:
Hi yes I checked the version of sed I wrote is indeed not portable, I believe your suggested change makes sense,

Reply:
> Could you also suggest how one could include tests for this?

You could parse out the version info from `deps/undici/src/package.json` and do `assert.strictEqual(process.versions.undici, expectedVersion)`

Reply:
> > Could you also suggest how one could include tests for this?
> 
> You could parse out the version info from `deps/undici/src/package.json` and do `assert.strictEqual(process.versions.undici, expectedVersion)`

Understood would test/tools be the correct directory for these tests?

Reply:
No that should be `test/parallel`